tests/e2e/vmgroups_test.go

```diff
@@ -431,3 +432,141 @@ func (vg *VmGroupTest) TestVmGroupVolumeClone(c *C) {
+// func (vg *VmGroupTest) TestVmgroupRemove(c *C) {
+// 	misc.LogTestStart(c.TestName())
+
+// 	Create a volume in vgTestVMgroup1
+// 	vg.createVolumes(c, vg.volName1)
+
+// 	Add create privilege for the second datastore
+// 	out, err := adminutils.AddCreateAccessForVMgroup(vg.config.EsxHost, vgTestVMgroup1, vg.config.Datastores[1])
+// 	c.Assert(err, IsNil, Commentf(out))
+
+// 	Create another volume on second datastore
+// 	vg.createVolumes(c, vg.volName2+"@"+vg.config.Datastores[1])
+
+// 	Verify volume can be mounted and unmounted for the first volume
+// 	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName1, vg.testContainer)
+// 	c.Assert(err, IsNil, Commentf(out))
+
+// 	Status should be detached
+// 	status := verification.VerifyDetachedStatus(vg.volName1, vg.config.Datastores[0], vg.config.DockerHosts[0], vg.config.EsxHost)
+// 	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", vg.volName1))
+
+// 	Verify volume can be mounted and unmounted for the second volume
+// 	out, err = dockercli.ExecContainer(vg.config.DockerHosts[0], vg.volName2+"@"+vg.config.Datastores[1], vg.testContainer)
+// 	c.Assert(err, IsNil, Commentf(out))
+
+// 	Status should be detached
+// 	status = verification.VerifyDetachedStatus(vg.volName2, vg.config.Datastores[1],
+// 		vg.config.DockerHosts[0], vg.config.EsxHost)
+// 	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", vg.volName2+"@"+vg.config.Datastores[1]))
+
+// 	remove VM1 and VM2 from vgTestVMgroup1 and then remove the vmgroup
+// 	vmList := []string{vg.config.DockerHostNames[0], vg.config.DockerHostNames[1]}
+// 	for _, vm := range vmList {
+// 		out, err = adminutils.RemoveVMFromVMgroup(vg.config.EsxHost, vgTestVMgroup1, vm)
+// 		c.Assert(err, IsNil, Commentf(out))
+// 	}
+
+// 	log.Printf("Removing test vmgroup %s", vgTestVMgroup1)
+// 	out, err = adminutils.DeleteVMgroup(vg.config.EsxHost, vgTestVMgroup1, false)
+// 	c.Assert(err, IsNil, Commentf(out))
+
+// 	volumeList := []string{vg.volName1, vg.volName2}
+// 	for _, volume := range volumeList {
+// 		volume does not belong to any vmgroup, so "out" is expected to be "N/A"
+// 		out, err = verification.GetVMGroupForVolume(vg.config.EsxHost, volume)
+// 		log.Println("GetVMGroupForVolume return out[%s] err[%s] for volume %s", out, err, volume)
+// 		c.Assert(out, Equals, "N/A", Commentf("volume %s should not belong to any vmgroup", volume))
+// 	}
+
+// 	Recreate the test VM group1 and add VM1, VM2 back to the vmgroup
+// 	cmd := adminconst.CreateVMgroup + vgTestVMgroup1 + " --default-datastore " + vg.config.Datastores[0]
+// 	log.Printf("Creating test vmgroup %s", vgTestVMgroup1)
+// 	out, err = ssh.InvokeCommand(vg.config.EsxHost, cmd)
+// 	c.Assert(err, IsNil, Commentf(out))
+
+// 	for _, vm := range vmList {
+// 		out, err = adminutils.AddVMToVMgroup(vg.config.EsxHost, vgTestVMgroup1, vm)
+// 		c.Assert(err, IsNil, Commentf(out))
+// 	}
```

Comment:
Makes sense to have a separate function to keep this code to restore the configuration.